lib/charms/lte_core_interface/v0/lte_core_interface.py

```diff
@@ -0,0 +1,246 @@
+from ops.main import main
+
+from charm.lte_core_interface.v0.lte_core_interface import (
+    CoreProvides,
+)
+
+
+class DummyCoreProviderCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.core_provider = CoreProvides(self, "lte-core")
+        self.framework.observe(
+            self.on.lte_core_relation_joined, self._on_lte_core_relation_joined
+        )
+
+    def _on_lte_core_relation_joined(self, event: RelationJoinedEvent) -> None:
+        if not self.unit.is_leader():
+            return
+        mme_ipv4_address = "some code for fetching the mme ipv4 address"
+        self.core_provider.set_core_information(mme_ipv4_address=mme_ipv4_address)
+
+
+if __name__ == "__main__":
+    main(DummyCoreProviderCharm)
+```
+
+"""
+
+import logging
+from ipaddress import AddressValueError, IPv4Address
+
+from jsonschema import exceptions, validate  # type: ignore[import]
+from ops.charm import CharmBase, CharmEvents, RelationChangedEvent
+from ops.framework import EventBase, EventSource, Handle, Object
+
+# The unique Charmhub library identifier, never change it
+LIBID = "3fbbdca922ec4ddd9598c3382034ad61"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+logger = logging.getLogger(__name__)
+
+REQUIRER_JSON_SCHEMA = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "`lte-core` requirer root schema",
+    "type": "object",
+    "description": "The `lte-core` root schema comprises the entire requirer databag for this interface.",  # noqa: E501
+    "examples": [
+        {
+            "mme_ipv4_address": "127.0.0.1",  # noqa: E501
+        }
+    ],
+    "properties": {
+        "mme_ipv4_address": {
+            "description": "The IP address of the MME (Mobility Management Entity) from core.",  # noqa: E501
+            "type": "string",
+            "format": "ipv4",
+        },
+    },
+    "required": [
+        "mme_ipv4_address",
+    ],
+    "additionalProperties": True,
+}
+
+
+class CoreAvailableEvent(EventBase):
+    """Charm event emitted when a core is available. It carries the mme ipv4 address."""
+
+    def __init__(self, handle: Handle, mme_ipv4_address: str):
+        """Init."""
+        super().__init__(handle)
+        self.mme_ipv4_address = mme_ipv4_address
+
+    def snapshot(self) -> dict:
+        """Returns snapshot."""
+        return {"mme_ipv4_address": self.mme_ipv4_address}
+
+    def restore(self, snapshot: dict) -> None:
+        """Restores snapshot."""
+        self.mme_ipv4_address = snapshot["mme_ipv4_address"]
+
+
+class CoreRequirerCharmEvents(CharmEvents):
+    """List of events that the core requirer charm can leverage."""
+
+    lte_core_available = EventSource(CoreAvailableEvent)
+
+
+class CoreRequires(Object):
```

Comment:
```suggestion
class LTECoreRequires(Object):
```

Reply:
This should be specifically LTE, if we support 5G in the future it'll likely be different.

Reply:
I've fixed the issue for all the missing "LTE"s.